docs/source-pytorch/advanced/model_parallel.rst

```diff
@@ -55,6 +55,94 @@ Sharding techniques help when model sizes are fairly large; roughly 500M+ parame
 
 ----------
 
+.. _Colossalai:
+
+**************************
+Colossalai
+**************************
+
```

Comment:
We could consider adding concise description here as the intro why or when a Lightning user should consider this strategy. Something along the lines of:

> Use when:
> - Training very large, billion parameter models
> - You need to save cost 
> - Train as big of a model you can given a fixed budget of hardware

This is just a dump example of course. But I would consider something like this before going into the technical part of how it works.